### PR TITLE
Imgur5k dataset integration

### PR DESCRIPTION
**Pull request by felixdittrich92**
Integrates IMGUR5K dataset

Todo: 
- [x] rework box computations

Closes:
#587

### CODE REVIEW

File: doctr/datasets/imgur5k.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            # some files have no annotations
+            if img_id not in annotation_file['index_to_ann_map'].keys():
+                continue
+            ann_ids = annotation_file['index_to_ann_map'][img_id]
+            annotations = [annotation_file['ann_id'][a_id] for a_id in ann_ids]
+
+            labels: List[str] = []
+            box_targets: List[np.ndarray] = []
+            for ann in annotations:
+                if ann['word'] != '.':
```
Comment (fg-mindee) [inline doctr/datasets/imgur5k.py:83]: Does that mean that the text "final." will have two boxes?

File: doctr/datasets/imgur5k.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                    labels.append(ann['word'])
+                    # x_center, y_center, width, height, angle
+                    _box = [float(val) for val in ann['bounding_box'].strip('[ ]').split(', ')]
+                    # (x, y) coordinates of top left, top right, bottom right, bottom left corners
+                    box_targets.append(cv2.boxPoints(((_box[0], _box[1]), (_box[2], _box[3]), _box[4])))
```
Comment (fg-mindee) [inline doctr/datasets/imgur5k.py:88]: Remember, list comprehension are much better :)
```python
labels = [ann['word'] for ann in annotations if ann['word'] != '.']
# x_center, y_center, width, height, angle
_boxes = [map(float, ann['bounding_box'].strip('[ ]').split(', ')) for ann in annotations if ann['word'] != '.']
# (x, y) coordinates of top left, top right, bottom right, bottom left corners
box_targets = [cv2.boxPoints(((box[0], box[1]), (box[2], box[3]), box[4])) for box in _boxes]
```

File: doctr/datasets/imgur5k.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+            if not use_polygons:
+                # xmin, ymin, xmax, ymax
+                box_targets = [np.array([min(box_target[:, 0]), min(box_target[:, 1]), max(
+                    box_target[:, 0]), max(box_target[:, 1])], dtype=np_dtype) for box_target in box_targets]
```
Comment (fg-mindee) [inline doctr/datasets/imgur5k.py:93]: box_target is already a np.ndarray
```python
box_targets = [np.concatenate((points.min(1), points.max(1)), axis=-1) for points in box_targets]
```

**Review by fg-mindee — COMMENTED** [review]
Thanks a lot Felix!
A few improvement suggestions in the comments but the rest is good!

File: doctr/datasets/imgur5k.py (hunk 1 above)
Comment (felixdittrich92) [reply doctr/datasets/imgur5k.py:83]: @fg-mindee 
no there are some wrong/missing annotations in the set which have only a dot in dict key 'word'
https://github.com/facebookresearch/IMGUR5K-Handwriting-Dataset/blob/main/README.md
Bounding boxes with '.' mean the annotations were not done for various reasons

Example from annotations:
```
"qNgR61W_99": {
            "word": ".",
            "bounding_box": "[479.0, 496.0, 45.0, 35.0, -3.0]"
        },
        "qNgR61W_100": {
            "word": ".",
            "bounding_box": "[529.0, 498.0, 52.0, 37.0, 0.0]"
        },
        "qNgR61W_101": {
            "word": ".",
            "bounding_box": "[583.0, 494.0, 42.0, 37.0, -2.0]"
        },
```

File: doctr/datasets/imgur5k.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+        for img_name in img_names[set_slice]:
+            img_path = Path(img_folder, img_name)
+            img_id = img_name.split(".")[0]
+
+            # File existence check
+            if not os.path.exists(os.path.join(self.root, img_name)):
+                raise FileNotFoundError(f"unable to locate {os.path.join(self.root, img_name)}")
+
+            # some files have no annotations
```
[the remark this answers is not in the document]
Comment (felixdittrich92) [reply doctr/datasets/imgur5k.py:74]: :+1:

File: doctr/datasets/imgur5k.py (hunk 1 above)
Comment (felixdittrich92) [reply doctr/datasets/imgur5k.py:83]: yep :)

File: doctr/datasets/imgur5k.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        >>> from doctr.datasets import IMGUR5K
+        >>> train_set = IMGUR5K(train=True, img_folder="/path/to/IMGUR5K-Handwriting-Dataset/images",
+        >>>                     label_path="/path/to/IMGUR5K-Handwriting-Dataset/dataset_info/imgur5k_annotations.json")
+        >>> img, target = train_set[0]
+        >>> test_set = IMGUR5K(train=False, img_folder="/path/to/IMGUR5K-Handwriting-Dataset/images",
+        >>>                    label_path="/path/to/IMGUR5K-Handwriting-Dataset/dataset_info/imgur5k_annotations.json")
+        >>> img, target = test_set[0]
+    Args:
+        img_folder: folder with all the images of the dataset
+        label_path: path to the annotations file of the dataset
```
[the remark this answers is not in the document]
Comment (fg-mindee) [reply doctr/datasets/imgur5k.py:37]: The arg descriptions need to be in the same order as in the `__init__` :upside_down_face:

**Review by fg-mindee — CHANGES_REQUESTED** [review]
Small typo and then we're good!

**Pull request merged**